core/main_test.ts

```diff
@@ -1113,12 +962,24 @@ actions:
   });
 });
 
+function runMainInVmForDirectory(projectDir: string): dataform.CoreExecutionResponse {
+  return runMainInVm(
+    dataform.CoreExecutionRequest.create({
+      compile: { compileConfig: { projectDir } }
+    })
+  );
+}
```

Comment:
this is fine if you want it, but i would suggest a more composition-friendly model.

```
function coreExecutionRequest(projectDir: string) {
  return dataform.CoreExecutionRequest.create({
    compile: { compileConfig: { projectDir } }
  });
}
```

```
runMainInVm(coreExecutionRequest(projectDir))
```